src/metabase/api/setup.clj

```diff
@@ -38,21 +38,33 @@
   (su/with-api-error-message (s/constrained su/NonBlankString setup/token-match?)
     "Token does not match the setup token."))
 
+(def ^:dynamic ^:private *disallow-api-setup-after-first-user-is-created*
```

Comment:
this should be `*allow-api-setup-after-first-user-created* false` and then the test can change this to `true` as needed.